contracts/contracts/vault/VaultAdmin.sol

```diff
@@ -343,6 +343,59 @@ contract VaultAdmin is VaultStorage {
         emit AssetSupported(_asset);
     }
 
+    function _removeAsset(address _asset) internal {
+        require(assets[_asset].isSupported, "Asset not supported");
+        require(
+            IVault(address(this)).checkBalance(_asset) == 0,
+            "Vault still holds asset"
+        );
+
+        uint256 assetsCount = allAssets.length;
+        uint256 assetIndex = assetsCount;
+        for (uint256 i = 0; i < assetsCount; ++i) {
+            if (allAssets[i] == _asset) {
+                assetIndex = i;
+                break;
```

Comment:
We should really revert after this loop if the asset is not found. In theory anything with isSupported will be the allAssets, but this could break stuff badly if it were not.

Reply:
Actually, `assetIndex` is set to `assetsCount` initially. So it'll revert if the asset doesn't exist `allAssets` with an out-of-bound index error. Have also added a comment about this (a few lines below this)

Reply:
Clever. I've pushed up one additional code comment on this.